### PR TITLE
WIP: Add example script to setup minikube with none driver on Debian

### PR DESCRIPTION
**Pull request by gustavomassa**
Installs minikube, docker, docker dependencies and kubectl.
Starts minikube using none-driver and minikube dashboard.

sudo chmod +x setup.sh
./setup.sh

Tested on Ubuntu Bionic and Linux Mint.

### CODE REVIEW

**Comment by medyagh** [comment]
I think a better place for this would be the docs ?

**Comment by gustavomassa** [comment]
> I think a better place for this would be the docs ?

Is up to you, I put on main folder because I did not see any scripts on the doc folder and there is already  a test.sh file on the main folder.

**Comment by afbjorklund** [comment]
This assumes that running minikube with the none driver on your laptop is something you want to do ?

It is currently **not** recommended, since there is no isolation between kubernetes and the rest of it.

**Comment by gustavomassa** [comment]
> This assumes that running minikube with the none driver on your laptop is something you want to do ?
> 
> It is currently **not** recommended, since there is no isolation between kubernetes and the rest of it.

It is not something that I want to, it is something I have to, because minikube needs network access to external databases where only the host has access, thus the easiest way is using none driver and hostNetwork options.

There are some TODOS on the script to pass the driver parameter and other important parameters to the script. The idea of the script is just to reduce the minikube installation dependency and make the life easier for everyone, the script is far away to be good or perfect, it is just a start, so everyone could help making it better.

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -e
```
Comment (tstromberg) [inline setup.sh]: Add short description and example usage here.

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+if [ "$EUID" -eq 0 ]
+  then echo "Run without sudo/root"
+  exit
+fi
+
+# Install Docker dependencies
+sudo apt-get -y install apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+sleep 1
```
Comment (tstromberg) [inline setup.sh]: sleep here is unneccessary.

File: setup.sh (hunk 1 above)
Comment (gustavomassa) [reply setup.sh]: done, thank you for the feedback. Feel free to improve the script.

File: setup.sh (hunk 2 above)
Comment (gustavomassa) [reply setup.sh]: Added --vm-drive params
Fixed minikube url to get the latest version
Removed sleeps
Added echo messages
Added kubectl context command

**Comment by gustavomassa** [comment]
> Cool script. Do you mind adding a README.md and moving this script to `contrib/examples/debian-none-installer/`?
> 
> It would be nice to cite this from our documentation page. Thank you!

You mean creating  /docs/contributors/examples folder? And adding the debian-installer.sh & README.md?

File: setup.sh (hunk 2 above)
Comment (gustavomassa) [reply setup.sh]: Should the script delete any previous versions of minikube before starting the new one?

Something like:
```
minikube delete
sudo rm -rf ~/.kube
sudo rm -rf ~/.minikube
sudo rm -rf /etc/kubernetes
```

**Comment by tstromberg** [comment]
> > Cool script. Do you mind adding a README.md and moving this script to `contrib/examples/debian-none-installer/`?
> > It would be nice to cite this from our documentation page. Thank you!
> 
> You mean creating /docs/contributors/examples folder? And adding the debian-installer.sh & README.md?

Correct.

**Comment by tstromberg** [comment]
Since this PR was raised, we've given each driver their own documentation page:

https://minikube.sigs.k8s.io/docs/reference/drivers/none/

Would you be willing to put this script in an examples subdirectory of this folder:

https://github.com/kubernetes/minikube/tree/master/site/content/en/docs/Reference/Drivers

**Comment by sharifelgamal** [comment]
Closing this for now. If you'd like to add this script as a tutorial or example in our driver documentation, feel free to open a new PR.

**Pull request closed without merging**